packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java

```diff
@@ -58,13 +59,54 @@ private Bundle convertArguments(Map<String, ?> arguments) {
         bundle.putDouble(key, (Double) value);
       } else if (value instanceof Long) {
         bundle.putLong(key, (Long) value);
+      } else if (value instanceof byte[]) {
+        bundle.putByteArray(key, (byte[]) value);
+      } else if (value instanceof int[]) {
+        bundle.putIntArray(key, (int[]) value);
+      } else if (value instanceof long[]) {
+        bundle.putLongArray(key, (long[]) value);
+      } else if (value instanceof double[]) {
+        bundle.putDoubleArray(key, (double[]) value);
+      } else if (isTypedArrayList(value, Integer.class)) {
```

Comment:
How does the PlatformChannel decide between sending int[] vs List\<Integer\> ?

Reply:
Use `typed_data` to send primitive arrays. E.g. `Int32List` maps to `int[]`.